cmd/doctor.go

```diff
@@ -74,6 +92,80 @@ func exePath() (string, error) {
 	return filepath.Abs(file)
 }
 
+func runDoctorPathInfo(ctx *cli.Context) ([]string, error) {
+
+	res := make([]string, 0, 10)
+
+	if fi, err := os.Stat(setting.CustomConf); err != nil || !fi.Mode().IsRegular() {
+		res = append(res, fmt.Sprintf("Failed to find configuration file at '%s'.", setting.CustomConf))
+		res = append(res, fmt.Sprintf("If you've never ran Gitea yet, this is normal and '%s' will be created for you on first run.", setting.CustomConf))
+		res = append(res, "Otherwise check that you are running this command from the correct path and/or provide a `--config` parameter.")
+		return res, fmt.Errorf("Can't proceed without a configuration file")
+	}
+
+	setting.NewContext()
+
+	fail := false
+
+	check := func(name, path string, is_dir, is_write bool) {
+		res = append(res, fmt.Sprintf("%-25s  '%s'", name+":", path))
+		if fi, err := os.Stat(path); err != nil {
+			res = append(res, fmt.Sprintf("    ERROR: %v", err))
+			fail = true
+			return
+		} else if is_dir && !fi.IsDir() {
+			res = append(res, "    ERROR: not a directory")
+			fail = true
+			return
+		} else if !is_dir && !fi.Mode().IsRegular() {
+			res = append(res, "    ERROR: not a regular file")
+			fail = true
+			return
+		} else if is_write {
+			if err := runDoctorWritableDir(path); err != nil {
+				res = append(res, fmt.Sprintf("    ERROR: not writable: %v", err))
+				fail = true
+				return
+			}
+		}
+		return
+	}
+
+	// Note print paths inside quotes to make any leading/trailing spaces evident
+	check("Configuration File Path", setting.CustomConf, false, false)
+	check("Repository Root Path", setting.RepoRootPath, true, true)
+	check("Data Root Path", setting.AppDataPath, true, true)
+	check("Custom File Root Path", setting.CustomPath, true, false)
```

Comment:
Since a `custom` folder isn't required to run Gitea, the message "ERROR" may be a little misleading.

Reply:
@jolheiser True that. I've changed it into `NOTICE: not accessible (msg)`.